pkg/codegen/schema/schema.go

```diff
@@ -1527,14 +1542,19 @@ func bindTypes(pkg *Package, complexTypes map[string]ComplexTypeSpec, loader Loa
 		typs.resources[r.Token] = &ResourceType{Token: r.Token}
 	}
 
-	// Process properties.
+	// Process enums.
```

Comment:
I resolve enums before processing object types because the default value resolution expects `ElementTypes` and `DefaultType` to be populated. This looks safe but also seems not ideal: I'm not a fan of this implicit dependency in processing order. Is there a better way to resolve enums on demand?

Reply:
There is already a processing order dependency in this function right above this section (https://github.com/pulumi/pulumi/blob/master/pkg/codegen/schema/schema.go#L1508-L1523). Seems reasonable to resolve the enums within that first loop rather than adding in an extra loop. It doesn't help with the order dependency but that is a pre-existing condition so you're not introducing it. I think clearly documenting the reasoning in a comment as we've done for declaring tokens above is sufficient.